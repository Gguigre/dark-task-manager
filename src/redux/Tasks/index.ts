export * from './actions';
export { tasksReducer } from './reducer';
