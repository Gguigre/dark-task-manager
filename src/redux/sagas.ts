import { SagaIterator } from 'redux-saga';
//import { all, fork } from 'redux-saga/effects';
//import { saga as authSaga } from './Auth';

export function* rootSaga(): SagaIterator {
  //yield all([fork(authSaga)]);
}
