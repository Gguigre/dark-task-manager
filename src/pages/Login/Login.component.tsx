import React, { FunctionComponent } from 'react';
import { Button, View, Text, ActivityIndicator, StyleSheet } from 'react-native';
import { NavigationInjectedProps } from 'react-navigation';
import I18nJS from 'i18n-js';
import { CheatCodesButton } from '../../components/CheatCodesButton';
import { useDispatch, useSelector } from 'react-redux';
import { isLoadingSelector } from '../../redux/LoadingStatus/selectors';
import { startLoading, finishLoading } from '../../redux/LoadingStatus';

const styles = StyleSheet.create({
  container: { flex: 1, alignItems: 'center', justifyContent: 'center' },
});
const LOADER_NAME = 'demoLoader';

export const Login: FunctionComponent<NavigationInjectedProps<{}>> = ({ navigation }) => {
  const dispatch = useDispatch();
  const isLoading = useSelector(isLoadingSelector(LOADER_NAME));

  const goToHomePage = (): void => {
    navigation.navigate('Home');
  };

  const toggleLoader = () => {
    if (!isLoading) {
      dispatch(startLoading(LOADER_NAME));
    } else {
      dispatch(finishLoading(LOADER_NAME));
    }
  };

  return (
    <View style={styles.container}>
      <Text>{I18nJS.t('login.title')}</Text>
      <Button title={I18nJS.t('login.homeButton')} onPress={goToHomePage} />
      <CheatCodesButton navigation={navigation} />
      <Text>{I18nJS.t('login.redux')}</Text>
      <Button title={I18nJS.t('login.loader')} onPress={toggleLoader} />
      {isLoading && <ActivityIndicator testID="demoLoader" />}
    </View>
  );
};
