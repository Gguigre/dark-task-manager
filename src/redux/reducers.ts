import { Action, combineReducers } from 'redux';
import { loadingStatusReducer, LoadingStatusState } from './LoadingStatus/reducer';

const appReducer = combineReducers({
  loadingStatus: loadingStatusReducer,
});

export interface RootState {
  loadingStatus: LoadingStatusState;
}

type RootReducer = ReturnType<typeof appReducer>;

export const rootReducer = (state: RootState | undefined, action: Action): RootReducer => {
  return appReducer(state, action);
};
