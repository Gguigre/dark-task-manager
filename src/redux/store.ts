import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { persistStore, persistReducer } from 'redux-persist';
import AsyncStorage from '@react-native-community/async-storage';
import { middlewares, sagaMiddleware } from './middlewares';

import { rootReducer } from './reducers';
import { rootSaga } from './sagas';

export const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  whitelist: [],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(
  persistedReducer,
  composeWithDevTools(applyMiddleware(...middlewares))
);
export const persistor = persistStore(store);

sagaMiddleware.run(rootSaga);
