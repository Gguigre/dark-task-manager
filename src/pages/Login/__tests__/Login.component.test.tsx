import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import I18nJS from 'i18n-js';
import { startLoading, finishLoading } from '../../../redux/LoadingStatus';
import { Login } from '../Login.component';

const mockStore = configureStore([]);
const initialState = { loadingStatus: {} };

describe('Login component', () => {
  const navigation = {
    navigate: jest.fn(),
  } as any; // eslint-disable-line @typescript-eslint/no-explicit-any

  it('should render correctly', () => {
    const store = mockStore(initialState);
    const login = render(
      <Provider store={store}>
        <Login navigation={navigation} />
      </Provider>
    );
    expect(login).toMatchSnapshot();
  });

  it('should have a button to go to the Home page', () => {
    const store = mockStore(initialState);
    const login = render(
      <Provider store={store}>
        <Login navigation={navigation} />
      </Provider>
    );

    const HomeButton = login.getByText(I18nJS.t('login.homeButton'));
    fireEvent.press(HomeButton);
    expect(navigation.navigate).toHaveBeenCalledWith('Home');
  });

  it('should trigger the loader', () => {
    const store = mockStore(initialState);
    const login = render(
      <Provider store={store}>
        <Login navigation={navigation} />
      </Provider>
    );

    expect(() => login.getByTestId('demoLoader')).toThrowError();

    const LoaderButton = login.getByText(I18nJS.t('login.loader'));
    fireEvent.press(LoaderButton);

    expect(store.getActions()).toEqual([startLoading('demoLoader')]);
  });

  it('should stop the loader', () => {
    const store = mockStore({ loadingStatus: { demoLoader: { isLoading: true } } });
    const login = render(
      <Provider store={store}>
        <Login navigation={navigation} />
      </Provider>
    );

    expect(login.getByTestId('demoLoader')).toBeDefined();

    const LoaderButton = login.getByText(I18nJS.t('login.loader'));
    fireEvent.press(LoaderButton);

    expect(store.getActions()).toEqual([finishLoading('demoLoader')]);
  });
});
