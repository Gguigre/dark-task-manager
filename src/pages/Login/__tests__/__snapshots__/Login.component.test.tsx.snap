// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Page de login
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Aller sur la home page
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        CheatCodes
      </Text>
    </View>
  </View>
  <Text>
    Redux test
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Toggle selector
      </Text>
    </View>
  </View>
</View>
`;
